File: back/src/main/java/com/back/global/initData/BaseInitData.java
package com.back.global.initData;

import com.back.domain.concert.entity.Concert;
import com.back.domain.concert.repository.ConcertRepository;
import com.back.domain.schedule.entity.Schedule;
import com.back.domain.venue.entity.Venue;
import com.back.global.initData.fixture.*;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Component
@Profile("!test")
@RequiredArgsConstructor
public class BaseInitData implements CommandLineRunner {

    private final ConcertRepository concertRepository;
    private final VenueFixture venueFixture;
    private final ConcertFixture concertFixture;
    private final ConcertDetailFixture concertDetailFixture;
    private final ScheduleFixture scheduleFixture;
    private final ScheduleSeatFixture scheduleSeatFixture;

    @Override
    @Transactional
    public void run(String... args) {
        if (concertRepository.count() > 0) {
            return;
        }

        List<Venue> venues = venueFixture.createVenues();
        List<Concert> concerts = concertFixture.createConcerts();
        concertDetailFixture.createDetails(concerts);
        List<Schedule> schedules = scheduleFixture.createSchedules(concerts, venues);
        scheduleSeatFixture.createSeats(schedules);
    }
}

File: back/src/main/kotlin/com/back/global/initData/BaseInitData.kt
package com.back.global.initData

import com.back.domain.concert.repository.ConcertRepository
import com.back.global.initData.fixture.*
import org.springframework.boot.CommandLineRunner
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Component
@Profile("!test")
class BaseInitData(
    private val concertRepository: ConcertRepository,
    private val venueFixture: VenueFixture,
    private val concertFixture: ConcertFixture,
    private val concertDetailFixture: ConcertDetailFixture,
    private val scheduleFixture: ScheduleFixture,
    private val scheduleSeatFixture: ScheduleSeatFixture
) : CommandLineRunner {

    @Transactional
    override fun run(vararg args: String) {
        if (concertRepository.count() > 0) {
            return
        }

        val venues = venueFixture.createVenues()
        val concerts = concertFixture.createConcerts()
        concertDetailFixture.createDetails(concerts)
        val schedules = scheduleFixture.createSchedules(concerts, venues)
        scheduleSeatFixture.createSeats(schedules)
    }
}
